spring-integration-cassandra/src/main/java/org/springframework/integration/cassandra/config/xml/CassandraParserUtils.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.integration.cassandra.config.xml;
+
+import java.util.List;
+
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.beans.factory.config.RuntimeBeanReference;
+import org.springframework.beans.factory.support.BeanDefinitionBuilder;
+import org.springframework.beans.factory.support.ManagedMap;
+import org.springframework.beans.factory.xml.ParserContext;
+import org.springframework.integration.config.ExpressionFactoryBean;
+import org.springframework.integration.config.xml.IntegrationNamespaceUtils;
+import org.springframework.util.CollectionUtils;
+import org.springframework.util.StringUtils;
+import org.springframework.util.xml.DomUtils;
+import org.w3c.dom.Element;
+
+/**
+ * @author Filippo Balicchia
+ *
+ */
+public class CassandraParserUtils {
+
+	public static void processOutboundTypeAttributes(Element element, ParserContext parserContext,
+			BeanDefinitionBuilder builder) {
+
+		String cassandraTemplate = element.getAttribute("cassandra-template");
+		String operationType = element.getAttribute("type");
+		String writeOptionsInstance = element.getAttribute("write-options");
+		String ingestQuery = element.getAttribute("cql-ingest");
+		String query = element.getAttribute("query");
+
+		if (StringUtils.isEmpty(cassandraTemplate)) {
+			parserContext.getReaderContext().error("cassandra-template is empty", element);
+		}
+
+		builder.addConstructorArgValue(new RuntimeBeanReference(cassandraTemplate));
+		if (!StringUtils.isEmpty(operationType)) {
+			builder.addConstructorArgValue(operationType);
+		}
+
+		if (!StringUtils.isEmpty(writeOptionsInstance)) {
+			builder.addPropertyReference("writeOptions", writeOptionsInstance);
+		}
+
+		if (!StringUtils.isEmpty(ingestQuery) && !"INSERT".equalsIgnoreCase(operationType)) {
```

Comment:
1. `ingestQuery` has a precedence and `queryType` is overridden in the setter. So, this check is redundant. And that's why I asked before do not have setter for the `queryType`.
2. We can't check the actual attribute value in the parser because we don't have it yet here. And that `operationType` maybe is just the PP or SpEL string. The actual value is extracted somewhere later on the Bean population phase.

Just remove and change the next `ingestQuery` population to the `IntegrationNamespaceUtils.setValueIfAttributeDefined()`!